src/CalcViewModel/ApplicationViewModel.cpp

```diff
@@ -125,6 +125,14 @@ bool ApplicationViewModel::TryRecoverFromNavigationModeFailure()
 void ApplicationViewModel::OnModeChanged()
 {
     assert(NavCategory::IsValidViewMode(m_mode));
+    if (NavCategory::IsValidViewMode(m_PreviousMode) && NavCategory::IsCalculatorViewMode(m_PreviousMode) && !NavCategory::IsCalculatorViewMode(m_mode))
+    {
+        if (m_CalculatorViewModel)
+        {
+            m_CalculatorViewModel->SetCalculatorType(m_mode);
+        }
+    }
+
```

Comment:
if you switch mode in Calculator (Programmer, Scientific, Standard) then the callback about mode (in this case "OperatorsPanel::EnsureProgrammerRadixOps()" is called.  But if you switch mode, not in Calculator, then the callback is not called because mode in Calculator is not changed. so I control the previous mode.

In my memory, Calculator value and the setting isn't changed if you switch mode, not in Calculator

but If you apply this, Calculator value and the setting is changed no matter which the previous mode you selected even it is not graph mode.

To apply only graph and calculator mode, we can change this "if statement". But I don't know what's right, please reply, Thank you!

-Previous Version
![previous-version](https://user-images.githubusercontent.com/10369528/89419427-51616a80-d76c-11ea-9d03-dd7225ef959d.gif)

-Now
![now](https://user-images.githubusercontent.com/10369528/89419451-558d8800-d76c-11ea-9f64-92eb1f38ffee.gif)

Reply:
Sorry, I don't think I was clear enough in my comment. I wasn't talking about the result or the state of the whole calculator resetting, I am talking about just the state of the BitShift flyout demonstrated in this GIF:

![flyout](https://user-images.githubusercontent.com/6334170/89563640-97501a00-d7d0-11ea-93b4-ba9dc4cdcd0a.gif)

However, after trying it out locally I noticed this still happens in master **before** your changes so it actually isn't an issue with this PR. Sorry for sidetracking you.

Reply:
I have reverted commit about this, Thanks for the comment!